### DISCUSSION TITLE
How the sequence.hooks = stack works?

### DISCUSSION THREAD

**Discussion by brunolopesr** [vitest-dev/vitest#6246] [Q&A]
Upgraded to Vitest 2.0 today and "something" changed in the order the beforeEach hooks are working and some tests that previously were not failing in 1.6.0, started to fail.

Changed the `sequence.hooks` to `parallel` and everything started to work correctly again.

So my question is, what is the order for the "before" hooks in the `stack`?

The parent hooks run after the test suite before hooks? Is it a last in, first out or I'm missing something?

**Comment by sheremet-va** [comment]
Before hooks are running one after another. After hooks are running in a reverse order
